tangelo/toolboxes/molecular_computation/molecule.py

```diff
@@ -299,65 +280,22 @@ def mo_coeff(self):
         Returns:
             np.array: MO coefficient as a numpy array.
         """
-        return self.mean_field.mo_coeff
+        return self.solver.mo_coeff
 
     @mo_coeff.setter
     def mo_coeff(self, new_mo_coeff):
         # Asserting the new molecular coefficient matrix have the same dimensions.
         if self.uhf:
             assert len(new_mo_coeff) == 2, "Must provide [alpha mo_coeff, beta_mo_coeff]"
-            assert ((self.mean_field.mo_coeff[0].shape == new_mo_coeff[0].shape) and
-                    (self.mean_field.mo_coeff[1].shape == new_mo_coeff[1].shape)), \
+            assert ((self.solver.mo_coeff[0].shape == new_mo_coeff[0].shape) and
```

Comment:
We often let people provide mo_coeff as a `List[ndarray]`. This has been changed to 
```python3
        assert self.solver.mo_coeff.shape == (new_mo_coeff := np.array(new_mo_coeff)).shape, \
            f"The new molecular coefficients matrix has a {new_mo_coeff.shape}"\
            f" shape: expected shape is {self.solver.mo_coeff.shape}."
        self.solver.mo_coeff = new_mo_coeff
```
where I used the assignment of a Walrus operator so that `new_mo_coeff` is changed to an array in the assert line. If we don't want to modify what the users provide, this can be changed.